elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/hibernate/hql/SubCollectionFetchQueryBuilder.java

```diff
@@ -78,37 +78,43 @@ public Query build() {
                     + extractToOneMergeJoins(relationship.getChildType(), childAlias);
 
             //SELECT parent_children from Parent parent JOIN parent.children parent_children
-            Query q = session.createQuery(SELECT
-                            + childAlias
-                            + FROM
-                            + parentName + SPACE + parentAlias
-                            + JOIN
-                            + parentAlias + PERIOD + relationshipName + SPACE + childAlias
-                            + joinClause
-                            + WHERE
-                            + filterClause
-                            + " AND " + parentAlias + "=:" + parentAlias
-                            + SPACE
-                            + getSortClause(sorting)
-            );
+            String queryText = SELECT
+                    + childAlias
+                    + FROM
+                    + parentName + SPACE + parentAlias
+                    + JOIN
+                    + parentAlias + PERIOD + relationshipName + SPACE + childAlias
+                    + joinClause
+                    + WHERE
+                    + filterClause
+                    + " AND " + parentAlias + "=:" + parentAlias
+                    + SPACE
+                    + getSortClause(sorting);
+            setQueryString(queryText);
+            Query q = session.createQuery(queryText);
 
             supplyFilterQueryParameters(q, predicates);
             return q;
-        }).orElse(session.createQuery(SELECT
-                            + childAlias
-                            + FROM
-                            + parentName + SPACE + parentAlias
-                            + JOIN
-                            + parentAlias + PERIOD + relationshipName + SPACE + childAlias
-                            + getJoinClauseFromSort(sorting)
-                            + extractToOneMergeJoins(relationship.getChildType(), childAlias)
-                            + " WHERE " + parentAlias + "=:" + parentAlias
-                            + getSortClause(sorting)
-        ));
+        }).orElse(getQuery(childAlias, parentName, parentAlias, relationshipName));
 
         query.setParameter(parentAlias, relationship.getParent());
 
         addPaginationToQuery(query);
         return query;
     }
+
+    private Query getQuery(String childAlias, String parentName, String parentAlias, String relationshipName) {
```

Comment:
Why break this out as a separate functioN?

Reply:
We want to create the query from the querytext (session.createQuery) and set the value of the member variable from this querytext. Cannot do both inside the .orElse statement, hence created a function that does this. Cannot use lambda expressions as Query is not a functional intreface.